oioioi/programs/templates/programs/report-signal-hint.html

```diff
@@ -0,0 +1,79 @@
+                <br>
+                - A failed assertion when using <code>assert</code> - commonly used by the C++ Standard Library
+                <br>
+                - Other illegal memory operations.
+            {% endblocktrans %}
+        {% elif signal == 7 %}
+            {% blocktrans %}
+                <b>The SIGBUS signal (signal 7)</b> can mean a general memory violation, and can
+                be caused by trying to access an invalid memory address.
+            {% endblocktrans %}
+        {% elif signal == 8 %}
+            {% blocktrans %}
+                <b>Most common causes of the SIGFPE signal (signal 8):</b>
+                <br>
+                - Overflows/underflows for floating-point numbers
+                <br>
+                - Division by zero, either with floating-point numbers and integers
+                <br>
+                - Other illegal arithmetic operations, e.g. square-root of negative numbers.
+            {% endblocktrans %}
+        {% elif signal == 9 %}
+            {% blocktrans %}
+                <b>Most common sources for the SIGKILL signal (signal 9):</b>
+                <br>
+                - Your global variable exceeds the memory limit
+                <br>
+                (The difference compared to MLE is that the program exits before it can even "fully start")
+                <br>
+                - Other unexpected operations causing the program to get terminated preemptively.
+            {% endblocktrans %}
+        {% elif signal == 11 %}
+            {% blocktrans %}
+                <b>Most common causes of the SIGSEGV signal (signal 11):</b>
+                <br>
+                - Out-of-bounds array accesses
+                <br>
+                - De-referencing an invalid pointer (an empty "null-pointer" or a pointer to memory which has already been freed)
+                <br>
+                - Other memory violation errors
+            {% endblocktrans %}
+        {% elif signal == 31 %}
+            {% blocktrans %}
+                <b>Most common causes of the SIGSYS signal (signal 31):</b>
+                <br>
+                - Using a system call (syscall) that's not allowed by this task/contest.
+                <br>
+                - Using an invalid/non-existent system call.
+            {% endblocktrans %}
+        {% else %}
+            <span style="color: red;font-size: 1.5rem;font-weight: bold">
+                {% blocktranslate with signal=signal %}
+                    Signal {{ signal }} shouldn't show up while checking...
+                    <u>We recommend reporting this issue.</u>
+                {% endblocktranslate %}
+            </span>
+        {% endif %}
+        <br>
+    {% endfor %}
+    <hr>
+    {% blocktranslate %}
+        Program execution can differ depending on the operating system used and the execution environment.
+        Because of this, we recommend using our Test runs.
+    {% endblocktranslate %}
```

Comment:
Many contests (or interactive tasks) don't have test runs, so I would suggest adding "(if available)" at the end.
Also, maybe using Linux could be recommended here too?

Reply:
Good point, I'll remove the 2nd line for now.
As for recommending Linux, ideally maybe this note could link to the given contest's "technical rules", [like we have for OI](https://oi.edu.pl/static/attachment/20240129/31oi_etap2_ustalenia_techniczne.pdf), but I think OI/OIJ are the only contests that explicitly declare that, and contestants should be fully aware of them before they could ever reach this message.